assignments/prototypes.js

```diff
@@ -41,70 +118,93 @@
     },
-    hp: 15,
+    hp: dice(6, 5),
     name: 'Sir Mustachio',
     faction: 'The Round Table',
-    weapons: [
-      'Giant Sword',
-      'Shield',
-    ],
-    language: 'Common Toungue',
-  });
-
-  const archer = new Humanoid({
+    weapons: ['Giant Sword', 'Shield'],
+    language: 'Common Toungue'
+});
+
+const archer = new Humanoid({
     createdAt: new Date(),
     dimensions: {
-      length: 1,
-      width: 2,
-      height: 4,
+        length: 1,
+        width: 2,
+        height: 4
     },
-    hp: 10,
+    hp: dice(6, 4),
     name: 'Lilith',
     faction: 'Forest Kingdom',
-    weapons: [
-      'Bow',
-      'Dagger',
-    ],
-    language: 'Elvish',
-  });
-
-  console.log(mage.createdAt); // Today's date
-  console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
-  console.log(swordsman.hp); // 15
-  console.log(mage.name); // Bruce
-  console.log(swordsman.faction); // The Round Table
-  console.log(mage.weapons); // Staff of Shamalama
-  console.log(archer.language); // Elvish
-  console.log(archer.greet()); // Lilith offers a greeting in Elvish.
-  console.log(mage.takeDamage()); // Bruce took damage.
-  console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+    weapons: ['Bow', 'Dagger'],
+    language: 'Elvish'
+});
+
+const paladin = new Hero({
+    createdAt: new Date(),
+    dimensions: {
+        length: 1,
+        width: 3,
+        height: 6
+    },
+    hp: dice(6, 5),
+    name: 'Eragon',
+    faction: 'Alagaesian Rebels',
+    weapons: ['Dragon Heart Crystal'],
+    language: 'Common Tongue'
+});
+
+const shade = new Villain({
+    createdAt: new Date(),
+    dimensions: {
+        length: 1,
+        width: 1,
+        height: 6
+    },
+    hp: dice(6, 3),
+    name: 'Durza',
+    faction: 'Alagaesian Loyalists',
+    weapons: ['Demon Soul'],
+    language: 'Common Tongue'
+});
+
+console.log(mage.createdAt); // Today's date
+console.log(archer.dimensions); // { length: 1, width: 2, height: 4 }
+console.log(swordsman.hp); // 15
+console.log(mage.name); // Bruce
+console.log(swordsman.faction); // The Round Table
+console.log(mage.weapons); // Staff of Shamalama
+console.log(archer.language); // Elvish
+console.log(archer.greet()); // Lilith offers a greeting in Elvish.
+console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
 
-  // Stretch task: 
-  // * Create Villian and Hero constructor functions that inherit from the Humanoid constructor function.  
-  // * Give the Hero and Villians different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
-  // * Create two new objects, one a villian and one a hero and fight it out with methods!
\ No newline at end of file
+while (paladin.hp > 0 && shade.hp > 0) {
+    console.log(paladin.paladinAttack(shade, 0));
+    if (shade.hp > 0) {
+        console.log(shade.shadeAttack(paladin, 0));
+    }
+}
```

Comment:
I love the `dice` function you've used throughout and the battle loop you implemented here!